issue 1501 and various improvements on unzipping code, importing

iniju opened the pull request:
progress messages, fixes on import media bugs

A general close inspection of the import code. There were multiple bugs on the media import (written earlier by me for 2.0.1, but never really tested, he, he), should be working better now, importing files when needed, skipping if the file already exists in collection.media folder and looks the same (only the first 1024 bytes are compared for speed reasons and of course only when filenames match).

Much better feedback to the user, as this operation can take a while.

iniju committed:
issue 1501 and various improvements on unzipping code, importing
progress messages, fixes on import media bugs - updated according to
feedback2

flerda commented:
I have a few comments above. Let me know if they make sense and if you think we could update the code.

iniju commented:
Code updated, summary of changes:
- Removed unused unzip method.
- Added somestream.close() in finally, but please have a look at it. I
  don't really like try/catch inside finally, even more so done thousands of
  times within a loop, so I put the whole for loop in a single try catch and
  made sure finally had close() calls in case we exit at a throw.
- Added the column names in SQL calls.

Cheers,
Kostas

On Mon, Jan 28, 2013 at 11:05 PM, Flavio Lerda notifications@github.comwrote:

> I have a few comments above. Let me know if they make sense and if you
> think we could update the code.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/flerda/Anki-Android/pull/37#issuecomment-12807931.

flerda commented:
I have a few more comments on this. Sorry for being a bit slow...

iniju commented:
You're working very late! I'm at a training today, so I'll process them
later.

Thanks for the review
On Jan 29, 2013 6:20 AM, "Flavio Lerda" notifications@github.com wrote:

> I have a few more comments on this. Sorry for being a bit slow...
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/flerda/Anki-Android/pull/37#issuecomment-12821012.

flerda commented on src/com/ichi2/libanki/Utils.java:
```diff
@@ -719,34 +718,65 @@ public static String convertStreamToString(InputStream is) {
 
-    public static boolean unzip(String filename, String targetDirectory) {
+    public static boolean unzipFiles(ZipFile zipFile, String targetDirectory, String[] zipEntries, HashMap<String, String> zipEntryToFilenameMap) {
+        byte[] buf = new byte[2048];
```
This should also be FILE_COPY_BUFFER_SIZE.

flerda commented:
A couple of really tiny comments this time around.
